core/templates/services/math-interactions.service.ts

```diff
@@ -81,11 +82,23 @@ export class MathInteractionsService {
         'Your answer has two symbols next to each other: "' + symbol1 +
         '" and "' + symbol2 + '".');
     }
+    if (
+      errorMessage === 'Cannot read property \'parent\' of undefined.') {
+      let invalidFunctionName = '';
+      for (let functionName of this.mathFunctionNames) {
+        if (expressionString.includes(functionName + '()')) {
+          invalidFunctionName = functionName;
+          break;
+        }
+      }
+      errorMessage = (
+        'The ' + invalidFunctionName +
+        ' function cannot be empty. Please enter a variable/number in it.');
+    }
     return errorMessage;
   }
 
-  _validateExpression(
-      expressionString: string, validVariablesList: string[]): boolean {
```

Comment:
Why is validVariablesList removed now?

Reply:
This is a general function that is a common validator for algebraic and numeric expressions. The validations specific to each kind (like checking valid variables in algebraic expressions) are carried out in their respective functions `validateAlgebraicExpression` and `validateNumericExpression`. This parameter was no longer needed in this general function so I've removed it.

Reply:
Ah ok.